dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/business/HostAPI.java

```diff
@@ -289,6 +290,21 @@
 	 */
 	public Host resolveHostName(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException ;
 
+	/**
+	 * This method takes a server name (from a web request) and maps it to a host.
+	 * It is designed to do a lightweight cache lookup to get the mapping from server name -> host
+	 * and to prevent unnecessary lucene lookups.
+	 * If does not exists a host with that serverName then null is returned
+	 * @param serverName
+	 * @param user
+	 * @param respectFrontendRoles
+	 * @return
+	 * @throws DotDataException
+	 * @throws DotSecurityException
+	 */
+	public Optional<Host> resolveHostNameWithoutDefault(String serverName, User user, boolean respectFrontendRoles) throws DotDataException, DotSecurityException ;

```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary modifier 'public' on method 'resolveHostNameWithoutDefault': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=4718001)